### PR TITLE
Nv stream events pyfunc callbacks

### LINKED ISSUE #61: Add callbacks for stream events to the pipeline and components
It will allow users to add custom reaction to the creation of a new stream (adding a new source), resetting a stream, and end-of-stream events (e.g. to reset a tracker).
1. Add bindings for [gst-nvevent.h](https://docs.nvidia.com/metropolis/deepstream/sdk-api/gst-nvevent_8h.html) (deepstream/lib/libnvdsgst_helper.so). There are some event helpers in savant/deepstream/utils.py, refactor code.
2. Add callback(s) for pyfunc and pipeline (NvDsPipeline) on stream start, reset, and end

### PR DESCRIPTION
**Pull request by denisvmedyantsev**

### CODE REVIEW

**Comment by bwsw** [comment]
@denisvmedyantsev 

please update the benchmarks table before marking PR as ready to review.
please add to Jetson NX information on what power mode/core mode we are using when running performance benchmarks.

File: Makefile [hunk 1]
```diff
@@ -19,8 +19,8 @@ build:
 	--build-arg SAVANT_VERSION=$(SAVANT_VERSION) \
 	--build-arg DEEPSTREAM_VERSION=$(DEEPSTREAM_VERSION) \
 	-f docker/$(DOCKER_FILE) \
-	-t savant-deepstream$(PLATFORM_SUFFIX) \
-	-t savant-deepstream$(PLATFORM_SUFFIX):$(SAVANT_VERSION)-$(DEEPSTREAM_VERSION) .
+	-t savant-deepstream$(PLATFORM_SUFFIX) .
+	#docker tag savant-deepstream$(PLATFORM_SUFFIX) savant-deepstream$(PLATFORM_SUFFIX):$(SAVANT_VERSION)-$(DEEPSTREAM_VERSION)
```
Comment (abramov-oleg) [inline Makefile]: Are these commented out `docker tag` commands useful?

File: gst_plugins/python/pyfunc.py [hunk 2]
```diff
@@ -119,14 +119,8 @@ def do_stop(self):
...
-        self.pyfunc.on_sink_event(event)
-        # return self.srcpad.push_event(event)
-        return GstBase.BaseTransform.do_sink_event(self, event)
-
-    def do_src_event(self, event: Gst.Event) -> bool:
-        """Do on src event."""
-        self.pyfunc.on_src_event(event)
-        return GstBase.BaseTransform.do_src_event(self, event)
+        self.pyfunc.on_event(event)
+        return self.srcpad.push_event(event)
```
Comment (abramov-oleg) [inline gst_plugins/python/pyfunc.py:123]: Is it functionally the same as calling the handler from the base class?

What's the reason for the change?

File: Makefile (hunk 1 above)
Comment (denisvmedyantsev) [reply Makefile]: So far, I don't see it. Deleted it.

**Review by abramov-oleg — APPROVED** [review]
